src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceAuthzStep.java

```diff
@@ -51,10 +47,15 @@ public StepResult doStep(FlightContext flightContext) throws RetryException {
   public StepResult undoStep(FlightContext flightContext) {
     FlightMap inputMap = flightContext.getInputParameters();
     // Only delete the Sam resource if we actually created it in the do step.
-    FlightMap workingMap = flightContext.getWorkingMap();
-    if (workingMap.get(AUTHZ_COMPLETED_KEY, Boolean.class)) {
-      UUID workspaceID = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    UUID workspaceID = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    try {
       samService.deleteWorkspace(userReq.getRequiredToken(), workspaceID);
+    } catch (SamApiException ex) {
+      // Do nothing if the resource to delete is not found, this may not be the first time undo is
```

Comment:
same here - it could be worth logging the do-nothing case to see how often this happens.